### PR TITLE
idf.py --resolved before reset option issue  (IDFGH-1925)

### PR DESCRIPTION
**Pull request by tatsutaigu**
https://github.com/espressif/esp-idf/issues/4092#issue-495540213
idf.py ignores menuconfig --> Serial flasher config --> before flashing (No reset) option.
this idf.py is resloved before reset option issue

### CODE REVIEW

File: tools/idf.py [hunk 1]
```diff
@@ -340,6 +340,9 @@ def _get_esptool_args(args):
     with open(os.path.join(args.build_dir, "flasher_args.json")) as f:
         flasher_args = json.load(f)
 
+    extra_esptool_args = flasher_args["extra_esptool_args"]
+    result += ["--before", extra_esptool_args["before"]]
+
     extra_esptool_args = flasher_args["extra_esptool_args"]
```
Comment (dobairoland) [inline tools/idf.py]: This extra line of `extra_esptool_args = flasher_args["extra_esptool_args"]` should be removed.

Comment (tatsutaigu) [reply tools/idf.py]: I removed extra_esptool_args = flasher_args["extra_esptool_args"] online:343.
Please check it.

**Comment by dobairoland** [comment]
Thank you @tatsutaigu.

@kumekay I think we can squash the commits and add the "Closes" and "Merges" link.

**Comment by kumekay** [comment]
@tatsutaigu thank you for contribution once again,
your changes were merged, since changes were squashed commit ids has changed, so github didn't handle it automatically, if you are ok with changes, could you please close this PR?

**Pull request closed without merging**